### PR TITLE
Add rule to update IRENA renewables capacities

### PR DESCRIPTION
**Pull request by tgi-climact**
This PR add an automated rule to retrieve data from IRENA. This was useful to get updated values for renewables capacities. This PR has updated values up to 2022 (previously 2018).

## Checklist

- [x] I tested my contribution locally and it seems to work fine.
- [x] Code and workflow changes are sufficiently documented.
- [x] Changed dependencies are added to `envs/environment.yaml`.
- [x] Changes in configuration options are added in all of `config.default.yaml`.
- [x] Changes in configuration options are also documented in `doc/configtables/*.csv`.
- [x] A release note `doc/release_notes.rst` is added.

### CODE REVIEW

File: rules/retrieve.smk [hunk 1]
```diff
@@ -39,6 +39,27 @@ if config["enable"]["retrieve"] and config["enable"].get("retrieve_databundle",
             "../scripts/retrieve_databundle.py"
 
 
+if config["enable"].get("retrieve_irena", True):
```
Comment (fneum) [inline rules/retrieve.smk:42]:
```suggestion
if config["enable"].get("retrieve_irena"):
```
to be consistent with default set in config

File: scripts/retrieve_irena.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        columns=["Technology"]
+    )
+    df_onwind = irena[irena.Technology.str.contains("Onshore")].drop(
+        columns=["Technology"]
+    )
+    df_pv = irena[irena.Technology.str.contains("Solar")].drop(columns=["Technology"])
+
+    df_offwind.to_csv(snakemake.output[0])
+    df_onwind.to_csv(snakemake.output[1])
+    df_pv.to_csv(snakemake.output[2])
```
Comment (fneum) [inline scripts/retrieve_irena.py:107]:
```suggestion
    df_offwind.to_csv(snakemake.output["offwind"])
    df_onwind.to_csv(snakemake.output["onwind"])
    df_pv.to_csv(snakemake.output["solar"])
```

**Review by fneum — APPROVED** [review]
Looks good, just some minor suggestions.

If the link is somewhat reliable, one could even set `retrieve_irena: true` by default and completely deleted "data/existing_infrastructure/*_IRENA.csv" files from repository.

**Comment by tgi-climact** [comment]
> Looks good, just some minor suggestions.
> 
> If the link is somewhat reliable, one could even set `retrieve_irena: true` by default and completely deleted "data/existing_infrastructure/*_IRENA.csv" files from repository.

I guess so. It's a fixed query directly provided by IRENA. Please note that the query is configured to **not** update time periods.

**Comment by fneum** [comment]
Thanks @tgi-climact for this nice PR!

**Comment by tgi-climact** [comment]
> Thanks @tgi-climact for this nice PR!

You're welcome @fneum !

**Pull request merged**